torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -30,9 +31,31 @@ namespace torch {
 namespace jit {
 
 char const* toString(OpCode op);
-
 namespace {
 
+struct tensor_value_hash {
+  std::size_t operator()(const at::Tensor& tensor) const {
+    std::stringstream tensor_stream;
+    tensor_stream << tensor;
+    std::string tensor_str = tensor_stream.str();
+    std::size_t h1 = std::hash<std::string>{}(tensor_str);
+    return h1;
+  }
+};
+
+struct tensor_value_equal {
+  bool operator()(const at::Tensor& a, const at::Tensor& b) const {
+    std::stringstream a_stream;
+    a_stream << a;
+    std::string a_str = a_stream.str();
+
+    std::stringstream b_stream;
+    b_stream << b;
+    std::string b_str = b_stream.str();
+    return a_str == b_str;
```

Comment:
Why isn't the standard operator==() on Tensor sufficient here?

Reply:
The hash function uses the string representation of Tensor as the key. To be consistent, it's better to use the string representation for equal comparison.